File: LAB QUIZZES/CSE221_Quiz01/_A1F/_A1F_Solution.java
import java.util.*;

public class _A1F_Solution {
    public static class Pair<K, V> {
        public K key;
        public V value;

        public Pair(K key, V value) {
            this.key = key;
            this.value = value;
        }

        public String toString() {
            return "(" + key + ", " + value + ")";
        }
    }

    public static Pair<Integer, ArrayList<Pair<Integer, Integer>>> solve(ArrayList<Integer> sIDs,
            ArrayList<Integer> sMarks) {
        int N = sIDs.size(), cnt = 0;
        ArrayList<Pair<Integer, Integer>> arr = new ArrayList<>();
        // Place your code here
        return new Pair<>(cnt, arr);
    }
}

File: LAB QUIZZES/CSE221_Quiz01/_A1F/_A1F_Solution.py
def solve(sIDs, sMarks):
    N = len(sIDs)
    cnt = 0
    arr = []
    for i in range(N):
        max_idx = i
        max = sMarks[i]
        flag = False
        for j in range(i+1, N):
            if sMarks[j] > max:
                max_idx = j
                max = sMarks[j]
                flag = True
            elif sMarks[j] == max:
                if sIDs[j]<sIDs[max_idx]:
                    max_idx = j
                    flag = True
        if flag == True:
            sMarks[i], sMarks[max_idx] = sMarks[max_idx], sMarks[i]
            sIDs[i], sIDs[max_idx] = sIDs[max_idx], sIDs[i]
            cnt += 1  
    for i, j in zip(sIDs, sMarks):
        id, mark = i, j
        arr.append((id, mark))
    return (cnt, arr)